src/pm/hydra/tools/bootstrap/external/external_common_launch.c

```diff
@@ -48,8 +48,7 @@ static HYD_status lsf_get_path(char **path)
         *path = MPL_strdup(HYDT_bsci_info.launcher_exec);
 
     if (*path == NULL) {
-        MPL_env2str("LSF_BINDIR", (const char **) &bin_dir);
-        if (bin_dir) {
+        if (MPL_env2str("LSF_BINDIR", (const char **) &bin_dir) && bin_dir) {
```

Comment:
This check looks redundant to me. `MPL_env2str` is already checking for non-NULL when it returns 1.

Reply:
I think you're right. I'll drop that part.